php/api/class-update-fonts.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @package MaterialDesign
+ */
+
+/**
+ * Class Google Fonts API .
+ *
+ * @package MaterialDesign
+ */
+
+namespace MaterialDesign\Plugin\Api;
+
+use Exception;
+use stdClass;
+use function json_decode;
+use function MaterialDesign\Plugin\get_plugin_instance;
+
+/**
+ * Class Update_Fonts
+ *
+ * @package MaterialDesign\Plugin\Api
+ */
+class Update_Fonts extends API_Base {
+
+	const TRANSIENT = 'google-fonts-json';
+
+	/**
+	 * Holds the Google Fonts API key.
+	 *
+	 * @var string|null
+	 */
+	protected $api_key;
+
+	/**
+	 * Whether to force loading from HTTP.
+	 *
+	 * @var bool
+	 */
+	public $force_http;
+
+	/**
+	 * Update_Fonts constructor.
+	 *
+	 * @param bool   $force_http Whether to force loading from HTTP. Defaults to false.
+	 * @param string $api_key    Manually pass in a Google API key. Defaults to false.
+	 */
+	public function __construct( $force_http = false, $api_key = false ) {
+		parent::__construct();
+
+		$this->force_http = $force_http;
+
+		if ( ! empty( $api_key ) ) {
+			$this->api_key = $api_key;
+		} elseif ( defined( 'GOOGLE_FONTS_API_KEY' ) && false !== GOOGLE_FONTS_API_KEY ) {
+			$this->api_key = GOOGLE_FONTS_API_KEY;
+		} else {
+			$this->api_key = null;
+		}
+
+		if ( empty( $this->api_key ) ) {
+			_material_design_error( '_material_design_no_apikey_textonly', $this->material_design_no_apikey() );
```

Comment:
Looks like `_material_design_no_apikey_textonly` function is not defined. But this would be redundant when we fallback to the key provided with the plugin.